### PR TITLE
Add elastic logging for reference data logs

### PR DESCRIPTION
**Pull request by michaelcheah**
**What this PR does / why we need it**:
This PR adds elasticsearch logging for reference data in the request logger component. 

Reference data can be sent as cloud events to the request logger, which will parse and insert the data exactly like it would with live predictions, with the only differences being that:
1.  it inserts it into an index with pattern `reference-log-<deployment_type>-<deployment_namespace>-<deployment_name>`
2. it does not insert the entire payload into every instances document in elasticsearch

This PR also introduces a fix for catching bad payloads i.e. if metadata can be found for a request, but the payload does not match the metadata, this throws a `BadPayloadException`, which is caught, logs a warning, and **does not** insert the data into elasticsearch.

Previously, if the size of payload columns were larger than the number of metadata features, no errors are thrown and it would be inserted. If payload columns were less than metadata features, it would throw a runtime error ([here](https://github.com/SeldonIO/seldon-core/blob/13a5b1ea5a4a962d03965908e132aa4154ee9761/components/seldon-request-logger/app/default_logger.py#L577) or [here](https://github.com/SeldonIO/seldon-core/blob/13a5b1ea5a4a962d03965908e132aa4154ee9761/components/seldon-request-logger/app/default_logger.py#L589))

**Special notes for your reviewer**:
A new index pattern has been introduced, so the elastic token needs the required permissions

**Does this PR introduce a user-facing change?**:

```release-note
Changed seldon-request-logger component to catch and not insert requests whose metadata do not match the payload dimensions
```

An example of data inserted into elasticsearch using a reference dataset from the tabular income dataset [in this example](https://docs.seldon.io/projects/alibi-detect/en/stable/examples/cd_chi2ks_adult.html) is shown here:
image

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **majolo**
You can assign the PR to them by writing `/assign @majolo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/seldon-request-logger/app/default_logger.py
Comment (SachinVarghese) [inline components/seldon-request-logger/app/default_logger.py:134]: what is added_content in this case?

**Review by SachinVarghese — COMMENTED** [review]
/lgtm

**Review by SachinVarghese — APPROVED** [review]

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[SachinVarghese](https://github.com/SeldonIO/seldon-core/pull/3566#pullrequestreview-746128083 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [SachinVarghese]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3566]: [Method Not Allowed]

**Pull request merged**
